<a href="https://colab.research.google.com/github/Zjmgpk/CS5530/blob/main/Assignment2_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [2]:
import numpy
import pandas
import sklearn

In [12]:
df = pandas.read_csv("/content/drive/MyDrive/Colab Notebooks/Assignment 2/Assignment 2_1/Raw_Data/train.csv")
df = df.drop('Unnamed: 0', axis=1)
df

Name    Location  Year  Kilometers_Driven  \
0     Hyundai Creta 1.6 CRDi SX Option        Pune  2015              41000   
1                         Honda Jazz V     Chennai  2011              46000   
2                    Maruti Ertiga VDI     Chennai  2012              87000   
3      Audi A4 New 2.0 TDI Multitronic  Coimbatore  2013              40670   
4               Nissan Micra Diesel XV      Jaipur  2013              86999   
...                                ...         ...   ...                ...   
5842                  Maruti Swift VDI       Delhi  2014              27365   
5843          Hyundai Xcent 1.1 CRDi S      Jaipur  2015             100000   
5844             Mahindra Xylo D4 BSIV      Jaipur  2012              55000   
5845                Maruti Wagon R VXI     Kolkata  2013              46000   
5846             Chevrolet Beat Diesel   Hyderabad  2011              47000   

     Fuel_Type Transmission Owner_Type     Mileage   Engine      Power  Seats  \
0       Diesel       Manual      First  19.67 kmpl  1582 CC  126.2 bhp    5.0   
1       Petrol       Manual      First    13 km/kg  1199 CC   88.7 bhp    5.0   
2       Diesel       Manual      First  20.77 kmpl  1248 CC  88.76 bhp    7.0   
3       Diesel    Automatic     Second   15.2 kmpl  1968 CC  140.8 bhp    5.0   
4       Diesel       Manual      First  23.08 kmpl  1461 CC   63.1 bhp    5.0   
...        ...          ...        ...         ...      ...        ...    ...   
5842    Diesel       Manual      First   28.4 kmpl  1248 CC     74 bhp    5.0   
5843    Diesel       Manual      First   24.4 kmpl  1120 CC     71 bhp    5.0   
5844    Diesel       Manual     Second   14.0 kmpl  2498 CC    112 bhp    8.0   
5845    Petrol       Manual      First   18.9 kmpl   998 CC   67.1 bhp    5.0   
5846    Diesel       Manual      First  25.44 kmpl   936 CC   57.6 bhp    5.0   

      New_Price  Price  
0           NaN  12.50  
1     8.61 Lakh   4.50  
2           NaN   6.00  
3           NaN  17.74  
4           NaN   3.50  
...         ...    ...  
5842  7.88 Lakh   4.75  
5843        NaN   4.00  
5844        NaN   2.90  
5845        NaN   2.65  
5846        NaN   2.50  

[5847 rows x 13 columns]

In [13]:
df_cleaned = df.dropna(subset=['New_Price'])
display(df_cleaned)

Name Location  Year  Kilometers_Driven  \
1                          Honda Jazz V  Chennai  2011              46000   
5     Toyota Innova Crysta 2.8 GX AT 8S   Mumbai  2016              36000   
8                      Maruti Ciaz Zeta    Kochi  2018              25692   
13          Mitsubishi Pajero Sport 4X4    Delhi  2014             110000   
18                    BMW 3 Series 320d    Kochi  2014              32982   
...                                 ...      ...   ...                ...   
5827              Tata Bolt Revotron XT  Chennai  2016              10000   
5830      Volkswagen Vento 1.6 Highline   Mumbai  2011              38000   
5833           Maruti Vitara Brezza VDi     Pune  2016              37208   
5838               Honda Brio 1.2 VX MT    Delhi  2013              33746   
5842                   Maruti Swift VDI    Delhi  2014              27365   

     Fuel_Type Transmission Owner_Type     Mileage   Engine       Power  \
1       Petrol       Manual      First    13 km/kg  1199 CC    88.7 bhp   
5       Diesel    Automatic      First  11.36 kmpl  2755 CC   171.5 bhp   
8       Petrol       Manual      First  21.56 kmpl  1462 CC  103.25 bhp   
13      Diesel       Manual      First   13.5 kmpl  2477 CC  175.56 bhp   
18      Diesel    Automatic      First  22.69 kmpl  1995 CC     190 bhp   
...        ...          ...        ...         ...      ...         ...   
5827    Petrol       Manual      First  17.57 kmpl  1193 CC    88.7 bhp   
5830    Petrol       Manual      First  16.09 kmpl  1598 CC   103.5 bhp   
5833    Diesel       Manual      First   24.3 kmpl  1248 CC    88.5 bhp   
5838    Petrol       Manual      First   18.5 kmpl  1198 CC    86.8 bhp   
5842    Diesel       Manual      First   28.4 kmpl  1248 CC      74 bhp   

      Seats   New_Price  Price  
1       5.0   8.61 Lakh   4.50  
5       8.0     21 Lakh  17.50  
8       5.0  10.65 Lakh   9.95  
13      7.0  32.01 Lakh  15.00  
18      5.0  47.87 Lakh  18.55  
...     ...         ...    ...  
5827    5.0   7.77 Lakh   4.00  
5830    5.0  11.91 Lakh   3.25  
5833    5.0   9.93 Lakh   7.43  
5838    5.0   6.63 Lakh   3.20  
5842    5.0   7.88 Lakh   4.75  

[815 rows x 13 columns]

I dropped the NaN of the New_Price column due beacause of how few values there were in the column. With there only being 815 out of 5847, it wouldnt have been possible to fill each of the missing values.

In [14]:
filtered_df = df_cleaned[(df_cleaned['Kilometers_Driven'] == 50000) & (df_cleaned['Fuel_Type'] == 'Electric')]
display(filtered_df)

Name Location  Year  Kilometers_Driven Fuel_Type  \
4313  Mahindra E Verito D4  Chennai  2016              50000  Electric   

     Transmission Owner_Type Mileage Engine   Power  Seats   New_Price  Price  
4313    Automatic      First     NaN  72 CC  41 bhp    5.0  13.58 Lakh   13.0

In [15]:
rows_with_nan = df_cleaned[df_cleaned.isnull().any(axis=1)]
display(rows_with_nan)

Name Location  Year  Kilometers_Driven Fuel_Type  \
4313  Mahindra E Verito D4  Chennai  2016              50000  Electric   

     Transmission Owner_Type Mileage Engine   Power  Seats   New_Price  Price  
4313    Automatic      First     NaN  72 CC  41 bhp    5.0  13.58 Lakh   13.0

In [16]:
df_cleaned = df_cleaned.drop(4313)
display(df_cleaned)

Name Location  Year  Kilometers_Driven  \
1                          Honda Jazz V  Chennai  2011              46000   
5     Toyota Innova Crysta 2.8 GX AT 8S   Mumbai  2016              36000   
8                      Maruti Ciaz Zeta    Kochi  2018              25692   
13          Mitsubishi Pajero Sport 4X4    Delhi  2014             110000   
18                    BMW 3 Series 320d    Kochi  2014              32982   
...                                 ...      ...   ...                ...   
5827              Tata Bolt Revotron XT  Chennai  2016              10000   
5830      Volkswagen Vento 1.6 Highline   Mumbai  2011              38000   
5833           Maruti Vitara Brezza VDi     Pune  2016              37208   
5838               Honda Brio 1.2 VX MT    Delhi  2013              33746   
5842                   Maruti Swift VDI    Delhi  2014              27365   

     Fuel_Type Transmission Owner_Type     Mileage   Engine       Power  \
1       Petrol       Manual      First    13 km/kg  1199 CC    88.7 bhp   
5       Diesel    Automatic      First  11.36 kmpl  2755 CC   171.5 bhp   
8       Petrol       Manual      First  21.56 kmpl  1462 CC  103.25 bhp   
13      Diesel       Manual      First   13.5 kmpl  2477 CC  175.56 bhp   
18      Diesel    Automatic      First  22.69 kmpl  1995 CC     190 bhp   
...        ...          ...        ...         ...      ...         ...   
5827    Petrol       Manual      First  17.57 kmpl  1193 CC    88.7 bhp   
5830    Petrol       Manual      First  16.09 kmpl  1598 CC   103.5 bhp   
5833    Diesel       Manual      First   24.3 kmpl  1248 CC    88.5 bhp   
5838    Petrol       Manual      First   18.5 kmpl  1198 CC    86.8 bhp   
5842    Diesel       Manual      First   28.4 kmpl  1248 CC      74 bhp   

      Seats   New_Price  Price  
1       5.0   8.61 Lakh   4.50  
5       8.0     21 Lakh  17.50  
8       5.0  10.65 Lakh   9.95  
13      7.0  32.01 Lakh  15.00  
18      5.0  47.87 Lakh  18.55  
...     ...         ...    ...  
5827    5.0   7.77 Lakh   4.00  
5830    5.0  11.91 Lakh   3.25  
5833    5.0   9.93 Lakh   7.43  
5838    5.0   6.63 Lakh   3.20  
5842    5.0   7.88 Lakh   4.75  

[814 rows x 13 columns]

Row 4313 was dropped due to no other row relating to it. Since there was no other row that was related to it in the car's model, I wouldn't be able to calculate a value for the milage.

In [17]:
df_cleaned['Mileage'] = df_cleaned['Mileage'].str.replace(' kmpl', '')
df_cleaned['Mileage'] = df_cleaned['Mileage'].str.replace(' km/kg', '')
df_cleaned['Engine'] = df_cleaned['Engine'].str.replace(' CC', '')
df_cleaned['Power'] = df_cleaned['Power'].str.replace(' bhp', '')
df_cleaned['New_Price'] = df_cleaned['New_Price'].str.replace(' Lakh', '')
df_cleaned['New_Price'] = df_cleaned['New_Price'].str.replace(' Cr', '')
df_cleaned

Name Location  Year  Kilometers_Driven  \
1                          Honda Jazz V  Chennai  2011              46000   
5     Toyota Innova Crysta 2.8 GX AT 8S   Mumbai  2016              36000   
8                      Maruti Ciaz Zeta    Kochi  2018              25692   
13          Mitsubishi Pajero Sport 4X4    Delhi  2014             110000   
18                    BMW 3 Series 320d    Kochi  2014              32982   
...                                 ...      ...   ...                ...   
5827              Tata Bolt Revotron XT  Chennai  2016              10000   
5830      Volkswagen Vento 1.6 Highline   Mumbai  2011              38000   
5833           Maruti Vitara Brezza VDi     Pune  2016              37208   
5838               Honda Brio 1.2 VX MT    Delhi  2013              33746   
5842                   Maruti Swift VDI    Delhi  2014              27365   

     Fuel_Type Transmission Owner_Type Mileage Engine   Power  Seats  \
1       Petrol       Manual      First      13   1199    88.7    5.0   
5       Diesel    Automatic      First   11.36   2755   171.5    8.0   
8       Petrol       Manual      First   21.56   1462  103.25    5.0   
13      Diesel       Manual      First    13.5   2477  175.56    7.0   
18      Diesel    Automatic      First   22.69   1995     190    5.0   
...        ...          ...        ...     ...    ...     ...    ...   
5827    Petrol       Manual      First   17.57   1193    88.7    5.0   
5830    Petrol       Manual      First   16.09   1598   103.5    5.0   
5833    Diesel       Manual      First    24.3   1248    88.5    5.0   
5838    Petrol       Manual      First    18.5   1198    86.8    5.0   
5842    Diesel       Manual      First    28.4   1248      74    5.0   

     New_Price  Price  
1         8.61   4.50  
5           21  17.50  
8        10.65   9.95  
13       32.01  15.00  
18       47.87  18.55  
...        ...    ...  
5827      7.77   4.00  
5830     11.91   3.25  
5833      9.93   7.43  
5838      6.63   3.20  
5842      7.88   4.75  

[814 rows x 13 columns]

In [18]:
df_cleaned = pandas.get_dummies(df_cleaned, columns=['Fuel_Type', 'Transmission'], dtype=int)
display(df_cleaned)

Name Location  Year  Kilometers_Driven  \
1                          Honda Jazz V  Chennai  2011              46000   
5     Toyota Innova Crysta 2.8 GX AT 8S   Mumbai  2016              36000   
8                      Maruti Ciaz Zeta    Kochi  2018              25692   
13          Mitsubishi Pajero Sport 4X4    Delhi  2014             110000   
18                    BMW 3 Series 320d    Kochi  2014              32982   
...                                 ...      ...   ...                ...   
5827              Tata Bolt Revotron XT  Chennai  2016              10000   
5830      Volkswagen Vento 1.6 Highline   Mumbai  2011              38000   
5833           Maruti Vitara Brezza VDi     Pune  2016              37208   
5838               Honda Brio 1.2 VX MT    Delhi  2013              33746   
5842                   Maruti Swift VDI    Delhi  2014              27365   

     Owner_Type Mileage Engine   Power  Seats New_Price  Price  \
1         First      13   1199    88.7    5.0      8.61   4.50   
5         First   11.36   2755   171.5    8.0        21  17.50   
8         First   21.56   1462  103.25    5.0     10.65   9.95   
13        First    13.5   2477  175.56    7.0     32.01  15.00   
18        First   22.69   1995     190    5.0     47.87  18.55   
...         ...     ...    ...     ...    ...       ...    ...   
5827      First   17.57   1193    88.7    5.0      7.77   4.00   
5830      First   16.09   1598   103.5    5.0     11.91   3.25   
5833      First    24.3   1248    88.5    5.0      9.93   7.43   
5838      First    18.5   1198    86.8    5.0      6.63   3.20   
5842      First    28.4   1248      74    5.0      7.88   4.75   

      Fuel_Type_Diesel  Fuel_Type_Petrol  Transmission_Automatic  \
1                    0                 1                       0   
5                    1                 0                       1   
8                    0                 1                       0   
13                   1                 0                       0   
18                   1                 0                       1   
...                ...               ...                     ...   
5827                 0                 1                       0   
5830                 0                 1                       0   
5833                 1                 0                       0   
5838                 0                 1                       0   
5842                 1                 0                       0   

      Transmission_Manual  
1                       1  
5                       0  
8                       1  
13                      1  
18                      0  
...                   ...  
5827                    1  
5830                    1  
5833                    1  
5838                    1  
5842                    1  

[814 rows x 15 columns]

In [19]:
df_cleaned['Car_Age'] = 2025 - df_cleaned['Year']
display(df_cleaned)

Name Location  Year  Kilometers_Driven  \
1                          Honda Jazz V  Chennai  2011              46000   
5     Toyota Innova Crysta 2.8 GX AT 8S   Mumbai  2016              36000   
8                      Maruti Ciaz Zeta    Kochi  2018              25692   
13          Mitsubishi Pajero Sport 4X4    Delhi  2014             110000   
18                    BMW 3 Series 320d    Kochi  2014              32982   
...                                 ...      ...   ...                ...   
5827              Tata Bolt Revotron XT  Chennai  2016              10000   
5830      Volkswagen Vento 1.6 Highline   Mumbai  2011              38000   
5833           Maruti Vitara Brezza VDi     Pune  2016              37208   
5838               Honda Brio 1.2 VX MT    Delhi  2013              33746   
5842                   Maruti Swift VDI    Delhi  2014              27365   

     Owner_Type Mileage Engine   Power  Seats New_Price  Price  \
1         First      13   1199    88.7    5.0      8.61   4.50   
5         First   11.36   2755   171.5    8.0        21  17.50   
8         First   21.56   1462  103.25    5.0     10.65   9.95   
13        First    13.5   2477  175.56    7.0     32.01  15.00   
18        First   22.69   1995     190    5.0     47.87  18.55   
...         ...     ...    ...     ...    ...       ...    ...   
5827      First   17.57   1193    88.7    5.0      7.77   4.00   
5830      First   16.09   1598   103.5    5.0     11.91   3.25   
5833      First    24.3   1248    88.5    5.0      9.93   7.43   
5838      First    18.5   1198    86.8    5.0      6.63   3.20   
5842      First    28.4   1248      74    5.0      7.88   4.75   

      Fuel_Type_Diesel  Fuel_Type_Petrol  Transmission_Automatic  \
1                    0                 1                       0   
5                    1                 0                       1   
8                    0                 1                       0   
13                   1                 0                       0   
18                   1                 0                       1   
...                ...               ...                     ...   
5827                 0                 1                       0   
5830                 0                 1                       0   
5833                 1                 0                       0   
5838                 0                 1                       0   
5842                 1                 0                       0   

      Transmission_Manual  Car_Age  
1                       1       14  
5                       0        9  
8                       1        7  
13                      1       11  
18                      0       11  
...                   ...      ...  
5827                    1        9  
5830                    1       14  
5833                    1        9  
5838                    1       12  
5842                    1       11  

[814 rows x 16 columns]

dplyr in Python:

1) Select:

In [20]:
selected_df = df_cleaned[['Name', 'Year']]
display(selected_df)

Name  Year
1                          Honda Jazz V  2011
5     Toyota Innova Crysta 2.8 GX AT 8S  2016
8                      Maruti Ciaz Zeta  2018
13          Mitsubishi Pajero Sport 4X4  2014
18                    BMW 3 Series 320d  2014
...                                 ...   ...
5827              Tata Bolt Revotron XT  2016
5830      Volkswagen Vento 1.6 Highline  2011
5833           Maruti Vitara Brezza VDi  2016
5838               Honda Brio 1.2 VX MT  2013
5842                   Maruti Swift VDI  2014

[814 rows x 2 columns]

2) Filter:

In [21]:
selected_df = df_cleaned[(df_cleaned['Kilometers_Driven'] > 100000) & (df_cleaned['Price'] > 10) & (df_cleaned['Car_Age']  > 10)]
display(selected_df)

Name   Location  Year  Kilometers_Driven  \
13    Mitsubishi Pajero Sport 4X4      Delhi  2014             110000   
913   Mitsubishi Pajero Sport 4X4  Bangalore  2013             132000   
996    Toyota Fortuner 2.8 4WD MT      Kochi  2013             105072   
2178   Toyota Fortuner 2.8 4WD MT  Hyderabad  2010             110000   
2211   Toyota Fortuner 2.8 4WD MT     Mumbai  2010             140000   
3525   Toyota Fortuner 2.8 4WD MT      Kochi  2011             169006   

     Owner_Type Mileage Engine   Power  Seats New_Price  Price  \
13        First    13.5   2477  175.56    7.0     32.01  15.00   
913       First    13.5   2477  175.56    7.0      34.6  12.49   
996       First   14.24   2755   174.5    7.0     36.69  13.28   
2178      First   14.24   2755   174.5    7.0     37.51  13.70   
2211      First   14.24   2755   174.5    7.0     37.46  10.75   
3525      First   14.24   2755   174.5    7.0     36.69  10.64   

      Fuel_Type_Diesel  Fuel_Type_Petrol  Transmission_Automatic  \
13                   1                 0                       0   
913                  1                 0                       0   
996                  1                 0                       0   
2178                 1                 0                       0   
2211                 1                 0                       0   
3525                 1                 0                       0   

      Transmission_Manual  Car_Age  
13                      1       11  
913                     1       12  
996                     1       12  
2178                    1       15  
2211                    1       15  
3525                    1       14

3) Rename:

In [22]:
renamed_df = df_cleaned.rename(columns={'Name': 'Model_Name', 'Seats':'No. of Seats'})
display(renamed_df)

Model_Name Location  Year  Kilometers_Driven  \
1                          Honda Jazz V  Chennai  2011              46000   
5     Toyota Innova Crysta 2.8 GX AT 8S   Mumbai  2016              36000   
8                      Maruti Ciaz Zeta    Kochi  2018              25692   
13          Mitsubishi Pajero Sport 4X4    Delhi  2014             110000   
18                    BMW 3 Series 320d    Kochi  2014              32982   
...                                 ...      ...   ...                ...   
5827              Tata Bolt Revotron XT  Chennai  2016              10000   
5830      Volkswagen Vento 1.6 Highline   Mumbai  2011              38000   
5833           Maruti Vitara Brezza VDi     Pune  2016              37208   
5838               Honda Brio 1.2 VX MT    Delhi  2013              33746   
5842                   Maruti Swift VDI    Delhi  2014              27365   

     Owner_Type Mileage Engine   Power  No. of Seats New_Price  Price  \
1         First      13   1199    88.7           5.0      8.61   4.50   
5         First   11.36   2755   171.5           8.0        21  17.50   
8         First   21.56   1462  103.25           5.0     10.65   9.95   
13        First    13.5   2477  175.56           7.0     32.01  15.00   
18        First   22.69   1995     190           5.0     47.87  18.55   
...         ...     ...    ...     ...           ...       ...    ...   
5827      First   17.57   1193    88.7           5.0      7.77   4.00   
5830      First   16.09   1598   103.5           5.0     11.91   3.25   
5833      First    24.3   1248    88.5           5.0      9.93   7.43   
5838      First    18.5   1198    86.8           5.0      6.63   3.20   
5842      First    28.4   1248      74           5.0      7.88   4.75   

      Fuel_Type_Diesel  Fuel_Type_Petrol  Transmission_Automatic  \
1                    0                 1                       0   
5                    1                 0                       1   
8                    0                 1                       0   
13                   1                 0                       0   
18                   1                 0                       1   
...                ...               ...                     ...   
5827                 0                 1                       0   
5830                 0                 1                       0   
5833                 1                 0                       0   
5838                 0                 1                       0   
5842                 1                 0                       0   

      Transmission_Manual  Car_Age  
1                       1       14  
5                       0        9  
8                       1        7  
13                      1       11  
18                      0       11  
...                   ...      ...  
5827                    1        9  
5830                    1       14  
5833                    1        9  
5838                    1       12  
5842                    1       11  

[814 rows x 16 columns]

4) Mutate

In [23]:
df_2 = df_cleaned.copy()
df_2['Price_Per_Kilometer'] = df_2['Price']/df_2['Kilometers_Driven']
display(df_2)

Name Location  Year  Kilometers_Driven  \
1                          Honda Jazz V  Chennai  2011              46000   
5     Toyota Innova Crysta 2.8 GX AT 8S   Mumbai  2016              36000   
8                      Maruti Ciaz Zeta    Kochi  2018              25692   
13          Mitsubishi Pajero Sport 4X4    Delhi  2014             110000   
18                    BMW 3 Series 320d    Kochi  2014              32982   
...                                 ...      ...   ...                ...   
5827              Tata Bolt Revotron XT  Chennai  2016              10000   
5830      Volkswagen Vento 1.6 Highline   Mumbai  2011              38000   
5833           Maruti Vitara Brezza VDi     Pune  2016              37208   
5838               Honda Brio 1.2 VX MT    Delhi  2013              33746   
5842                   Maruti Swift VDI    Delhi  2014              27365   

     Owner_Type Mileage Engine   Power  Seats New_Price  Price  \
1         First      13   1199    88.7    5.0      8.61   4.50   
5         First   11.36   2755   171.5    8.0        21  17.50   
8         First   21.56   1462  103.25    5.0     10.65   9.95   
13        First    13.5   2477  175.56    7.0     32.01  15.00   
18        First   22.69   1995     190    5.0     47.87  18.55   
...         ...     ...    ...     ...    ...       ...    ...   
5827      First   17.57   1193    88.7    5.0      7.77   4.00   
5830      First   16.09   1598   103.5    5.0     11.91   3.25   
5833      First    24.3   1248    88.5    5.0      9.93   7.43   
5838      First    18.5   1198    86.8    5.0      6.63   3.20   
5842      First    28.4   1248      74    5.0      7.88   4.75   

      Fuel_Type_Diesel  Fuel_Type_Petrol  Transmission_Automatic  \
1                    0                 1                       0   
5                    1                 0                       1   
8                    0                 1                       0   
13                   1                 0                       0   
18                   1                 0                       1   
...                ...               ...                     ...   
5827                 0                 1                       0   
5830                 0                 1                       0   
5833                 1                 0                       0   
5838                 0                 1                       0   
5842                 1                 0                       0   

      Transmission_Manual  Car_Age  Price_Per_Kilometer  
1                       1       14             0.000098  
5                       0        9             0.000486  
8                       1        7             0.000387  
13                      1       11             0.000136  
18                      0       11             0.000562  
...                   ...      ...                  ...  
5827                    1        9             0.000400  
5830                    1       14             0.000086  
5833                    1        9             0.000200  
5838                    1       12             0.000095  
5842                    1       11             0.000174  

[814 rows x 17 columns]

5) Arrange


In [24]:
arranged_df = df_cleaned.sort_values(by='Year', ascending=True)
display(arranged_df)

Name    Location  Year  Kilometers_Driven  \
4723               Hyundai Santro D Lite     Chennai  2001             140000   
5414               Honda CR-V Petrol 2WD     Chennai  2004             135000   
337                     Skoda Octavia RS      Jaipur  2005             100000   
2058      Ford Freestyle Titanium Petrol      Mumbai  2007               1261   
1323           Hyundai Verna CRDi 1.6 SX   Hyderabad  2008              96578   
...                                  ...         ...   ...                ...   
5644        Skoda Rapid 1.6 MPI Ambition  Coimbatore  2019              21214   
1160   Volkswagen Polo 1.0 MPI Trendline   Hyderabad  2019               1000   
1322  Hyundai Grand i10 1.2 Kappa Sportz       Kochi  2019              18453   
165     Hyundai Grand i10 1.2 Kappa Asta     Kolkata  2019               1000   
226       Toyota Innova Crysta 2.8 GX AT       Kochi  2019              14165   

     Owner_Type Mileage Engine   Power  Seats New_Price  Price  \
4723      Third    20.3   1086      68    5.0      4.55   0.70   
5414      Third    14.4   1997     152    5.0     33.51   3.00   
337       First   14.45   1984     227    5.0     28.37   2.05   
2058      First    19.0   1194   94.68    5.0      8.12   6.00   
1323      First    22.0   1582   126.2    5.0     14.07   2.50   
...         ...     ...    ...     ...    ...       ...    ...   
5644      First   15.41   1598  103.52    5.0     12.44   9.66   
1160      First   18.78    999      75    5.0      6.74   5.68   
1322      First    18.9   1197   81.86    5.0      6.68   6.17   
165       First    18.9   1197   81.86    5.0      7.39   6.50   
226       First   11.36   2755   171.5    7.0     20.74  19.64   

      Fuel_Type_Diesel  Fuel_Type_Petrol  Transmission_Automatic  \
4723                 0                 1                       0   
5414                 0                 1                       1   
337                  0                 1                       1   
2058                 0                 1                       0   
1323                 1                 0                       0   
...                ...               ...                     ...   
5644                 0                 1                       0   
1160                 0                 1                       0   
1322                 0                 1                       0   
165                  0                 1                       0   
226                  1                 0                       1   

      Transmission_Manual  Car_Age  
4723                    1       24  
5414                    0       21  
337                     0       20  
2058                    1       18  
1323                    1       17  
...                   ...      ...  
5644                    1        6  
1160                    1        6  
1322                    1        6  
165                     1        6  
226                     0        6  

[814 rows x 16 columns]

6) Summarize with Group By:

In [25]:
summarized_df = df_cleaned.groupby('Name')['Kilometers_Driven'].mean()
display(summarized_df)

Name
Audi A3 35 TDI Premium Plus           39000.000000
Audi A4 30 TFSI Premium Plus          61221.000000
Audi A4 35 TDI Premium Plus           33534.666667
Audi A4 35 TDI Technology             29500.000000
Audi A6 35 TDI Matrix                 43706.000000
                                          ...     
Volkswagen Vento 1.5 TDI Trendline    47000.000000
Volkswagen Vento 1.6 Comfortline      33576.500000
Volkswagen Vento 1.6 Highline         42143.666667
Volvo S60 D4 Momentum                 51000.000000
Volvo V40 D3 R Design                 38000.000000
Name: Kilometers_Driven, Length: 317, dtype: float64